src/api/Awards/streak.js

```diff
@@ -0,0 +1,52 @@
+
+function checkStreak(match, playerName, streaks) {
+    let streak = streaks[playerName];
+      
+    if (!streak) {
+        streaks[playerName] = {
+            count: 1,
+            type: match.winner === playerName ? 'winning' : 'losing',
+            isOver: false
+        };
+    } else if (streak.isOver) {
+        // do nothing
+    } else if (streak.type === 'winning' && match.winner === playerName) {
+        streak.count += 1;
+    } else if (streak.type === 'losing' && match.loser === playerName) {
+        streak.count += 1;
+    } else {
+        streak.isOver = true;
+    }
+}
+
+// Assumes match results are sorted with most recent games first
+function calculate(matchResults) {
+    const awards = [];
+    let streaks = {};
+    
```

Comment:
Oops, a and b are matchResults so the assignments to aDate and bDate should be like:

aDate = moment(a.played_on);

Or something :)